yarn-project/end-to-end/src/e2e_p2p/add_rollup.test.ts

```diff
@@ -0,0 +1,646 @@
+          },
+          content: contentOutFromRollup.toString() as Hex,
+        };
+
+        const leaf = sha256ToField([
+          testContract.address,
+          new Fr(rollupVersion), // aztec version
+          ethRecipient.toBuffer32(),
+          new Fr(clients.publicClient.chain.id), // chain id
+          contentOutFromRollup,
+        ]);
+
+        const [l2MessageIndex, siblingPath] = await node.getL2ToL1MessageMembershipWitness(
+          l2OutgoingReceipt!.blockNumber!,
+          leaf,
+        );
+
+        // We need to mark things as proven
+        const cheatcodes = CheatCodes.createRollup(l1RpcUrls, l1ContractAddresses);
+        await cheatcodes.markAsProven();
+
+        // Then we want to go and comsume it!
+        const outbox = getContract({
+          address: l1ContractAddresses.outboxAddress.toString(),
+          abi: OutboxAbi,
+          client: clients.walletClient,
+        });
+
+        const { receipt: txReceipt } = await l1TxUtils.sendAndMonitorTransaction({
+          to: outbox.address,
+          data: encodeFunctionData({
+            abi: OutboxAbi,
+            functionName: 'consume',
+            args: [
+              l2ToL1Message,
+              BigInt(l2OutgoingReceipt!.blockNumber!),
+              BigInt(l2MessageIndex),
+              siblingPath.toBufferArray().map((buf: Buffer) => `0x${buf.toString('hex')}`) as readonly `0x${string}`[],
+            ],
+          }),
+        });
+
+        // Exactly 1 event should be emitted in the transaction
+        expect(txReceipt.logs.length).toBe(1);
+
+        // We decode the event log before checking it
+        const txLog = txReceipt.logs[0];
+        const topics = decodeEventLog({
+          abi: OutboxAbi,
+          data: txLog.data,
+          topics: txLog.topics,
+        }) as {
+          eventName: 'MessageConsumed';
+          args: {
+            l2BlockNumber: bigint;
+            root: `0x${string}`;
+            messageHash: `0x${string}`;
+            leafIndex: bigint;
+          };
+        };
+
+        // We check that MessageConsumed event was emitted with the expected message hash and leaf index
+        expect(topics.args.messageHash).toStrictEqual(leaf.toString());
+        expect(topics.args.leafIndex).toStrictEqual(BigInt(0));
+      }
+    };
+
+    await briding(
+      nodes[0],
+      t.ctx.initialFundedAccounts[0],
+      {
+        walletClient: t.ctx.deployL1ContractsValues.walletClient,
+        publicClient: t.ctx.deployL1ContractsValues.publicClient,
+      },
+      t.ctx.deployL1ContractsValues.l1ContractAddresses,
+      BigInt(t.ctx.aztecNodeConfig.rollupVersion),
+      t.ctx.aztecNodeConfig.l1RpcUrls,
+    );
+
+    let govData;
+    while (true) {
+      govData = await govInfo();
+      if (govData.leaderVotes >= quorumSize) {
+        break;
+      }
+      await sleep(12000);
+    }
+
+    expect(govData.leaderVotes).toBeGreaterThan(govBefore.leaderVotes);
```

Comment:
why does this check matter? we already know we are at quorom to get to this point?